### PR TITLE
Backport the new pipeline to flatcar-lts-2605 branch

### PR DESCRIPTION
**Pull request by krnowak**
~The commits are still a mess, I'll recreate them before marking this PR as ready for review.~ I didn't cherry-pick all the commits that introduced the ci-automation - too much work.

Here I basically copied `ci-automation`, `sdk_lib` and `sdk_container` and backported some of the changes from the `main` branch (like compression stuff).

Should be merged together with https://github.com/flatcar-linux/coreos-overlay/pull/2039, https://github.com/kinvolk/jenkins-os/pull/246. (Actually, coreos-overlay needs to go first, so I can update submodules in this repo.)

CI: http://jenkins.infra.kinvolk.io:8080/job/container/job/sdk/146/cldsv/

### CODE REVIEW

**Comment by krnowak** [comment]
First two commits basically bring stuff from main branch.

Third commit sets up the submodules and version file.

Fourth commit is this old LTS-specific workaround for the fact that it uses quite an old version of bash.

Fifth commit could be something to port to the main branch. The problem that I have with this commit is which URL to use for the torcx packages - I used bincache for now, but I suppose it should be something more persistent.

**Review by jepio — APPROVED** [review]
Glanced over the first commit, reviewed the rest and seems OK other than one nit. Could we rebase this and rerun the pipeline to make sure it runs through correctly?

File: build_library/release_util.sh [hunk 1]
```diff
@@ -42,6 +40,110 @@ DEFINE_string sign "" \
...
+    # Files that did not match the filter for disk images.
+    local -n local_extra_files="$3"
+
+    info "Compressing images"
+    # We want to compress images, but we also want to remove the uncompressed files
+    # from the list of uploadable files.
+    for filename in "${local_files_to_evaluate[@]}"; do
+        if [[ "${filename}" =~ \.(img|bin|vdi|vhd|vmdk)$ ]]; then
+            local -a FORMATS
+            FORMATS=($(echo "${FLAGS_image_compression_formats}" | tr ',' ' '))
```
[the remark this answers is not in the document]
Comment (krnowak) [reply build_library/release_util.sh:102]: Hm, this clobbers IFS:
``` bash
$ arr=(w x y z)
$ echo "${arr[@]}"
w x y z
$ echo "${arr[*]}"
w x y z

$ stuff=a,b,c,d
$ IFS=, arr2=( $stuff )
$ echo "$IFS"
,
# I don't get why I'm getting the output below, though.
$ echo $IFS

$ echo ${#arr2[@]}
4
$ echo "${arr2[@]}"
a b c d
# Got commas here again
$ echo "${arr2[*]}"
a,b,c,d
$ IFS=' '
$ echo "${arr2[*]}"
a b c d
```

**Comment by krnowak** [comment]
> Could we rebase this and rerun the pipeline to make sure it runs through correctly?

Done, CI running now: http://jenkins.infra.kinvolk.io:8080/job/container/job/sdk/175/cldsv/

**Comment by krnowak** [comment]
Also, do we want https://github.com/flatcar-linux/scripts/pull/388/commits/c769c2bac6b1a599084f708fbb418e5f50c3d80b to be backported to main and other branches?

**Comment by krnowak** [comment]
> Also, do we want [c769c2b](https://github.com/flatcar-linux/scripts/commit/c769c2bac6b1a599084f708fbb418e5f50c3d80b) to be backported to main and other branches?

To answer myself - probably not. We want to phase torcx out, so we most likely will not be adding extra torcx packages to the new releases, especially ones that are not available in the production image (extra package).

**Comment by krnowak** [comment]
Failures:
- `cl.internet` on EM `s3.xlarge.x86`.
- `cl.internet` on GCE `gvnic`.
- `kubeadm.v1.24.1.*` everywhere, should not be run, addressed with https://github.com/flatcar-linux/mantle/pull/353

File: build_library/release_util.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jepio) [reply build_library/release_util.sh:102]: OK, too bad

**Comment by krnowak** [comment]
> > Also, do we want [c769c2b](https://github.com/flatcar-linux/scripts/commit/c769c2bac6b1a599084f708fbb418e5f50c3d80b) to be backported to main and other branches?
> 
> To answer myself - probably not. We want to phase torcx out, so we most likely will not be adding extra torcx packages to the new releases, especially ones that are not available in the production image (extra package).

Backported in anyway, to have a feature parity with the old pipeline, PR is #452.

**Comment by pothos** [comment]
Looks good, needs to be aligned with https://github.com/flatcar-linux/scripts/pull/452 when that is merged. I'll also have to port the release job PR.

**Comment by krnowak** [comment]
Yeah, I'll rebase it to include the torcx changes.

**Pull request merged**